src/agent/debugger/src/target.rs

```diff
@@ -567,8 +565,26 @@ impl Target {
         Ok(current_context.get_flags())
     }
 
-    pub fn read_memory(&self, remote_address: LPCVOID, buf: &mut [impl Copy]) -> Result<()> {
-        process::read_memory_array(self.process_handle, remote_address, buf)
+    pub fn read_memory(&mut self, remote_address: LPCVOID, buf: &mut [impl Copy]) -> Result<()> {
```

Comment:
WDYT about making this method itself `unsafe`?

Reply:
I'm not sure that would add value as it's unclear how far up the callstack to stop adding `unsafe` and I think this function is easily reviewed for safety concerns.

Reply:
My concern is that, among other things, the current API would let us create an invalid reference in safe Rust, for example, if `buf` had type `&mut [&[T]]`.

I think `unsafe` for this method is the right place to have it: it's on the caller to ensure that they are safely transmuting the underlying remote bytes to a valid Rust type.

Reply:
Opened https://github.com/microsoft/onefuzz/issues/815 to track.